src/App.tsx

```diff
@@ -17,18 +37,24 @@
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                query={query}
+                setQuery={setQuery}
+                selectedCondition={selectedCondition}
+                setSelectedCondition={setSelectedCondition}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+
+              {!loading && todos.length > 0 && (
+                <TodoList todos={filteredData} />
```

Comment:
You're generating keys on render, which can lead to performance issues and unexpected behavior in React. Keys should be stable between renders for optimal performance. You can read more about this issue [here](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052).